releasenotes/notes/fix-circuit-builder-registers-21deba9a43356fb5.yaml

```diff
@@ -0,0 +1,46 @@
+---
+fixes:
+  - |
+    The control-flow builder interface now supports using :class:`.ClassicalRegister`\ s
+    as conditions in nested control-flow scopes.  Previously, doing this would
+    not raise an error immediately, but the internal circuit blocks would not
+    have the correct registers defined, and so later logic that worked with the
+    inner blocks would fail.
+
+    For example, previously the drawers would fail when trying to draw an inner
```

Comment:
The drawers really seem to sniff out bugs. We should consider adding drawer tests of anything that changes DAG generation ^^.

Reply:
Kind of!  They try and draw as much as they can, so yeah, I suppose they're a pretty good way of hitting quite a lot of the circuit.

Reply:
Just as an FYI, there is also https://github.com/Qiskit/qiskit-terra/blob/5c4cd2bbcfaf32f4db112f76f8f13128872afde5/test/python/dagcircuit/test_dagcircuit.py#L45 which tries to cover the same problem (but requires regular upkeep to stay consistent with changes to `DAGCircuit`).